File: time_write.py
import time

def main():
    N = 10000  # Modify as needed
    output_file = "data/time_output.txt"

    start_time = time.time()

    # Simulate a loop doing computations
    total = sum(range(N))

    end_time = time.time()
    run_time = end_time - start_time

    # Write to a file
    with open(output_file, "w") as f:
        f.write(f"Time taken for N={N}: {run_time:.6f} seconds\n")

if __name__ == "__main__":
    main()

File: time_write.c
#include <stdio.h>
#include <time.h>

int main() {
    FILE *fptr;
    fptr = fopen("data/time_output.txt", "w"); // Open file for writing

    if (fptr == NULL) {
        printf("Error opening file!\n");
        return 1;
    }

    clock_t start_time, end_time;
    start_time = clock();  // Start time

    // Simulate a loop doing computations
    long total = 0;
    for (long i = 0; i < 1000000; i++) {
        total += i;
    }

    end_time = clock();  // End time
    double run_time = ((double)(end_time - start_time)) / CLOCKS_PER_SEC;

    // Write execution time to file
    fprintf(fptr, "Time taken: %f seconds\n", run_time);
    fclose(fptr);

    printf("Execution time written to file!\n");
    return 0;
}
